sdks/go/pkg/beam/core/runtime/xlangx/expansionx/download.go

```diff
@@ -0,0 +1,127 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package expansionx contains utilities for starting expansion services for
+// cross-language transforms.
+package expansionx
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+)
+
+// URL is a type used to differentiate between strings and URLs
+type URL string
```

Comment:
Is it necessary for this type to be exported? 

Can or should it be replaced with the [net/url.URL](https://pkg.go.dev/net/url@go1.17.4#URL) type instead? (Probably not, frankly)